test/pytest/test_bridge.py

```diff
@@ -419,6 +419,29 @@ async def test_fsread1_errors(transport):
                                reply_keys={'message': "attribute 'max_read_size': must have type int"})
 
 
+@pytest.mark.asyncio
+async def test_fsread1_size_hint(transport):
+    data = None
+    stat = os.stat('/etc/passwd')
+    with open('/etc/passwd', 'rb') as fp:
+        data = fp.read()
+    ch = await transport.check_open('fsread1', path='/etc/passwd', binary='raw',
+                                    reply_keys={'size-hint': stat.st_size})
+    await transport.assert_data(ch, data)
```

Comment:
This is going to break if `/etc/passwd` is big enough to get chunked.  Since you do separate testcases now (and don't worry about the extra messages after the thing you check), you could just leave this part out if you wanted.

Otherwise, if you really want to check it, I'd do the loop and count bytes, but not compare contents.  At the end, make sure it was equal to the size hint.

Reply:
So, read os-release instead?

Reply:
I was more thinking "don't write a test that depends on the specific size of the file", but I agree that os-release is quite unlikely to be over 16k.  If this breaks at some point, you get to fix it, though :)